subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -102,15 +102,64 @@ static inline bool isr_rx_ci_adva_check(uint8_t tx_addr, uint8_t *addr,
 #endif
 
 #define PDU_MEM_SIZE       PDU_ADV_MEM_SIZE
-#define PDU_MEM_COUNT_MIN  (BT_CTLR_ADV_SET + \
-			    (BT_CTLR_ADV_SET * PAYLOAD_FRAG_COUNT) + \
-			    (BT_CTLR_ADV_AUX_SET * PAYLOAD_FRAG_COUNT) + \
-			    (BT_CTLR_ADV_SYNC_SET * PAYLOAD_FRAG_COUNT))
-#define PDU_MEM_FIFO_COUNT ((BT_CTLR_ADV_SET * PAYLOAD_FRAG_COUNT * 2) + \
-			    (CONFIG_BT_CTLR_ADV_DATA_BUF_MAX * \
+
+/* AD data and Scan Response Data need 2 PDU buffers each in the double buffer
+ * implementation. Allocate 3 PDU buffers plus CONFIG_BT_CTLR_ADV_DATA_BUF_MAX
+ * defined 1 PDU buffer as the minimum number of buffers that meet the legacy
+ * advertising needs. Add 1 each for Extended and Periodic Advertising, needed
+ * extra for double buffers for these is kept as configurable, by increasing
+ * CONFIG_BT_CTLR_ADV_DATA_BUF_MAX.
+ */
```

Comment:
Nit-pick: Mentioning here, what is default value in CONFIG_BT_CTLR_ADV_DATA_BUF_MAX, will require update if that value changes. Maybe remove that?